pytm/pytm.py

```
@@ -1701,9 +1713,16 @@ def encode_threat_data(obj):
     for e in obj:
         t = copy.deepcopy(e)
 
+        if (isinstance(t, Finding)):
+            attrs.append("threat_id")
+
         for a in attrs:
             v = getattr(e, a)
-            setattr(t, a, html.escape(v))
+
+            if (isinstance(v, int)):
+                t._safeset(a, v)
+            else:
+                t._safeset(a, html.escape(v))
```

`_safeset` ignored errors from `var*` attributes when trying to assign a value where it has already been done once. It's only used when resolving the model to figure out defaults for attributes where users did not explicitly assigned values.

Here, we have full control over how Finding objects are created. It should not be necessary. Also, using it here won't have any effect if the values are already assigned.

---

Would it be sufficient to cast all attributes to strings? They're only used in reports anyway.

---

Do you mean html.escape(str(v)) ? That solves the int problem, but the escaping doesn't seem to be 'taking'.

---

Is there a test that fails on master? I'll take a look.

---

Ah we should just skip encoding ids all together, if they're ints. Remove `id` from `attrs`.

---

`Finding.id` should be a string.

---

Like so: https://github.com/izar/pytm/pull/158

---

_safeset was used because due to deepcopy the values are already set, this is resetting the values to encoded versions and throwing ValueError when using setattr.

I thought about using a str but decided against it the 'id' is an int so I wrote code to respect that. Making it a str will impact json output and if someone was performing numeric operations.